src/WithBackgroundImage/index.js

```diff
@@ -1,26 +1,70 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import PropTypes from 'prop-types';
 
 import useBaseUrl from '@docusaurus/useBaseUrl';
-import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 import useThemeContext from '@theme/hooks/useThemeContext';
 
+let counter = 0;
+
+const getImage = (
+  [imageDark, imageDarkHover, imageLight, imageLightHover],
+  isHovered,
+) => {
+  if (!imageLight) {
+    return '';
+  }
+
+  const {isDarkTheme} = useThemeContext();
+
+  const backgroundImage = isDarkTheme && imageDark ? imageDark : imageLight;
+
+  let hoverBackgroundImage = isDarkTheme && imageDarkHover ? imageDarkHover : imageLightHover;
+  hoverBackgroundImage = hoverBackgroundImage ? hoverBackgroundImage : backgroundImage;
+
+  let image = isHovered ? hoverBackgroundImage : backgroundImage;
+
+  return useBaseUrl(image);
+};
+
+// TODO: Simplify the passing in of images
```

Comment:
TODOs need you name in parens after :

TODO (oliver): comment

Reply:
Hmm. I never realized it but adding a name is a lot better of a system for tracking todos